fail to build dali with gcc 14 due to missing cstdint header

### Version

1.40.0

### Describe the bug.

https://github.com/NVIDIA/DALI/blob/v1.40.0/dali/util/uri.h#L51 use `uint32_t` without including the cstdint header, this leads to compilation failure.

### Minimum reproducible example

```shell
none
```


### Relevant log output

```shell
In file included from /build/python-nvidia-dali/src/python-nvidia-dali/dali/pipeline/workspace/workspace.h:27,
                 from /build/python-nvidia-dali/src/python-nvidia-dali/dali/pipeline/workspace/sample_workspace.cc:15:
/build/python-nvidia-dali/src/python-nvidia-dali/dali/pipeline/data/tensor_list.h:87:34: warning: template-id not allowed for constructor in C++20 [-Wtemplate-id-cdtor]
   87 |   DLL_PUBLIC TensorList<Backend>(TensorList<Backend> &&other) noexcept;
      |                                  ^~~~~~~~~~~~~~~~~~~
/build/python-nvidia-dali/src/python-nvidia-dali/dali/pipeline/data/tensor_list.h:87:34: note: remove the ‘< >’
[ 38%] Building CXX object dali/CMakeFiles/dali.dir/util/cufile.cc.o
In file included from /build/python-nvidia-dali/src/python-nvidia-dali/dali/util/uri.cc:15:
/build/python-nvidia-dali/src/python-nvidia-dali/dali/util/uri.h:51:8: error: use of enum ‘ParseOpts’ without previous declaration
   51 |   enum ParseOpts : uint32_t {
      |        ^~~~~~~~~
/build/python-nvidia-dali/src/python-nvidia-dali/dali/util/uri.h:51:20: error: ‘uint32_t’ was not declared in this scope
   51 |   enum ParseOpts : uint32_t {
      |                    ^~~~~~~~
/build/python-nvidia-dali/src/python-nvidia-dali/dali/util/uri.h:24:1: note: ‘uint32_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   23 | #include "dali/core/api_helper.h"
  +++ |+#include <cstdint>
   24 | 
/build/python-nvidia-dali/src/python-nvidia-dali/dali/util/uri.h:51:29: error: default member initializer for unnamed bit-field
   51 |   enum ParseOpts : uint32_t {
      |                             ^
```


### Other/Misc.

quick fix, add one line to https://github.com/NVIDIA/DALI/blob/v1.40.0/dali/util/uri.h#L17
```c++
#include <cstdint>
```

### Check for duplicates

- [X] I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report

--- a/dali/util/uri.h
+++ b/dali/util/uri.h
@@ -17,6 +17,7 @@
 
 #include <cassert>
 #include <cstring>
+#include <cstdint>
 #include <stdexcept>
 #include <string>
 #include <string_view>

